.gitignore

```diff
@@ -0,0 +1,142 @@
+### macOS ###
+# General
+.DS_Store
+.AppleDouble
+.LSOverride
+# Icon must end with two \r
+Icon
+# Thumbnails
+._*
+# Files that might appear in the root of a volume
+.DocumentRevisions-V100
+.fseventsd
+.Spotlight-V100
+.TemporaryItems
+.Trashes
+.VolumeIcon.icns
+.com.apple.timemachine.donotpresent
+# Directories potentially created on remote AFP share
+.AppleDB
```

Comment:
Are these gitignore items absolutely necessary? For the ones specific to your environment, could you include them in your desktop's global gitignore and only keep the necessary ones for this project in the repo? I think the only ones necessary for a VSCode environment are the `build` and `node_modules`. It might cause confusion for other developers who do not have these files on their desktop.